src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
+  static alive = [];
   // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  static removeDeadAnimals() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  bite(target) {
+    if (!(target instanceof Herbivore) || target.hidden) {
+      return;
+    }
+    target.health -= 50;
```

Comment:
```suggestion
    if (target instanceof Herbivore && !target.hidden) {
      target.health -= 50;
    }
```

we can simplify it a bit